File: src/LinkDisk/network/AppSettings.java
package LinkDisk.network;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.util.Properties;

public class AppSettings {

    private static final String SETTINGS_FILE = "linkdisk_settings.properties";

    private static final String KEY_RECEIVE_DIR = "receiveDir";

    private static final String DEFAULT_RECEIVE_DIR = "received_files";

    public static String getReceiveDir() {

        Properties properties = loadProperties();

        String receiveDir = properties.getProperty(KEY_RECEIVE_DIR);

        if (receiveDir == null || receiveDir.trim().length() == 0) {
            receiveDir = DEFAULT_RECEIVE_DIR;
        }

        File folder = new File(receiveDir);

        if (!folder.exists()) {
            folder.mkdirs();
        }

        return folder.getPath();
    }

    public static void setReceiveDir(String receiveDir) {

        if (receiveDir == null || receiveDir.trim().length() == 0) {
            return;
        }

        File folder = new File(receiveDir);

        if (!folder.exists()) {
            folder.mkdirs();
        }

        Properties properties = loadProperties();

        properties.setProperty(KEY_RECEIVE_DIR, folder.getAbsolutePath());

        saveProperties(properties);
    }

    private static Properties loadProperties() {

        Properties properties = new Properties();

        File file = new File(SETTINGS_FILE);

        if (!file.exists()) {
            return properties;
        }

        FileInputStream in = null;

        try {
            in = new FileInputStream(file);

            properties.load(in);

        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            closeQuietly(in);
        }

        return properties;
    }

    private static void saveProperties(Properties properties) {

        FileOutputStream out = null;

        try {
            out = new FileOutputStream(SETTINGS_FILE);

            properties.store(out, "LinkDisk Settings");

        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            closeQuietly(out);
        }
    }

    private static void closeQuietly(java.io.Closeable closeable) {

        try {
            if (closeable != null) {
                closeable.close();
            }
        } catch (Exception e) {
            // ignore
        }
    }
}

File: linkdisk_flutter/副本lib/network/AppSettings.dart
import 'dart:io';

class AppSettings {
  // 配置文件名、键、默认值（和 Java 完全一致）
  static const String _settingsFile = "linkdisk_settings.properties";
  static const String _keyReceiveDir = "receiveDir";
  static const String _defaultReceiveDir = "received_files";

  // 获取接收目录（自动创建文件夹）
  static String getReceiveDir() {
    final props = _loadProperties();
    String? receiveDir = props[_keyReceiveDir];

    // 空值使用默认目录
    if (receiveDir == null || receiveDir.trim().isEmpty) {
      receiveDir = _defaultReceiveDir;
    }

    // 自动创建目录
    final folder = Directory(receiveDir);
    if (!folder.existsSync()) {
      folder.createSync(recursive: true);
    }

    return folder.path;
  }

  // 设置接收目录（自动保存到配置文件）
  static void setReceiveDir(String receiveDir) {
    if (receiveDir.trim().isEmpty) return;

    // 自动创建目录
    final folder = Directory(receiveDir);
    if (!folder.existsSync()) {
      folder.createSync(recursive: true);
    }

    final props = _loadProperties();
    props[_keyReceiveDir] = folder.absolute.path;
    _saveProperties(props);
  }

  // 加载 properties 文件
  static Map<String, String> _loadProperties() {
    final file = File(_settingsFile);
    if (!file.existsSync()) return {};

    try {
      final lines = file.readAsLinesSync();
      return _parseProperties(lines);
    } catch (e) {
      print(e);
      return {};
    }
  }

  // 保存 properties 文件
  static void _saveProperties(Map<String, String> props) {
    final file = File(_settingsFile);
    try {
      final buffer = StringBuffer();
      buffer.writeln("# LinkDisk Settings");

      props.forEach((key, value) {
        buffer.writeln("$key=$value");
      });

      file.writeAsStringSync(buffer.toString());
    } catch (e) {
      print(e);
    }
  }

  // 解析 .properties 格式（支持注释、空行）
  static Map<String, String> _parseProperties(List<String> lines) {
    final map = <String, String>{};
    for (final line in lines) {
      final trimmed = line.trim();
      if (trimmed.isEmpty || trimmed.startsWith('#') || trimmed.startsWith('!')) {
        continue;
      }
      final index = trimmed.indexOf('=');
      if (index == -1) continue;
      final key = trimmed.substring(0, index).trim();
      final value = trimmed.substring(index + 1).trim();
      map[key] = value;
    }
    return map;
  }
}